### PR TITLE
Add documentation around field aliases.

### PR DESCRIPTION
**Pull request by jtibshirani**

### CODE REVIEW

File: docs/reference/mapping/types/alias.asciidoc [hunk 1]
```diff
@@ -0,0 +1,101 @@
+[[alias]]
+=== Alias datatype
+
+An `alias` mapping defines an alternate name for a field in the  index.
+The alias can be used in place of the target field in <<search, search>> requests,
+and select other APIs like <<search-field-caps, field capabilities>>.
```
Comment (cbuescher) [inline docs/reference/mapping/types/alias.asciidoc]: s/select/selected?

**Review by cbuescher — APPROVED** [review]
Great to read.

File: docs/reference/mapping.asciidoc [hunk 2]
```diff
@@ -124,8 +124,10 @@ fields to an existing index with the <<indices-put-mapping,PUT mapping API>>.
 
 Other than where documented, *existing field mappings cannot be
 updated*. Changing the mapping would mean invalidating already indexed
-documents.  Instead, you should create a new index with the correct mappings
-and <<docs-reindex,reindex>> your data into that index.
+documents. Instead, you should create a new index with the correct mappings
+and <<docs-reindex,reindex>> your data into that index. If you only wish
+to rename a field and not change its mappings, it may make sense to introduce
+an <<alias, `alias`>> field.
```
Comment (jpountz) [inline docs/reference/mapping.asciidoc]: I've noticed that we usually don't put spaces after the comma, not sure whether it has rendering implications.

File: docs/reference/mapping/types/alias.asciidoc [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+  }
+}
+--------------------------------
+// CONSOLE
+
+<1> The path to the target field. Note that this must be the full path, including any parent
+objects (e.g. `object1.object2.field`).
+
+All relevant components of the search request accept field aliases. In particular, aliases can be
```
Comment (jpountz) [inline docs/reference/mapping/types/alias.asciidoc]: Feels weird to say "all" when source filtering is an exception?

File: docs/reference/mapping/types/alias.asciidoc [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+--------------------------------
+GET /_search
+{
+    "query" : {
+        "match_all": {}
+    },
+    "_source": "route_length_miles"
+}
+--------------------------------
+// CONSOLE
```
Comment (jpountz) [inline docs/reference/mapping/types/alias.asciidoc]: let's add `TEST[continued]`?

File: docs/reference/mapping/types/alias.asciidoc [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+--------------------------------
+GET /_search
+{
+    "query" : {
+        "match_all": {}
+    },
+    "docvalue_fields": ["route_*", "transit_mode"]
+}
+--------------------------------
+// CONSOLE
```
Comment (jpountz) [inline docs/reference/mapping/types/alias.asciidoc]: let's add `TEST[continued]`?

**Review by jpountz — APPROVED** [review]
LGTM, I only left minor comments.

File: docs/reference/mapping/types/alias.asciidoc (hunk 1 above)
Comment (jtibshirani) [reply docs/reference/mapping/types/alias.asciidoc]: Oops, thanks!

File: docs/reference/mapping/types/alias.asciidoc (hunk 4 above)
Comment (jtibshirani) [reply docs/reference/mapping/types/alias.asciidoc]: 👍

File: docs/reference/mapping/types/alias.asciidoc (hunk 5 above)
Comment (jtibshirani) [reply docs/reference/mapping/types/alias.asciidoc]: 👍

File: docs/reference/mapping.asciidoc (hunk 2 above)
Comment (jtibshirani) [reply docs/reference/mapping.asciidoc]: I'll check -- thanks.

File: docs/reference/mapping/types/alias.asciidoc (hunk 3 above)
Comment (jtibshirani) [reply docs/reference/mapping/types/alias.asciidoc]: Will fix, I could see this being misleading.

File: docs/reference/mapping.asciidoc (hunk 2 above)
Comment (jtibshirani) [reply docs/reference/mapping.asciidoc]: Update: I couldn't observe any rendering difference. I'll plan to keep it as-is, I think the space makes this read more clearly.

**Pull request merged**